dashboard/src/main/java/org/openedx/courses/presentation/DashboardGalleryViewModel.kt

```diff
@@ -277,35 +286,39 @@ class DashboardGalleryViewModel(
 
     private fun detectUnfulfilledPurchase() {
         viewModelScope.launch(Dispatchers.IO) {
-            val enrolledCourses =
+            runCatching {
```

Comment:
According to current coding practices, we should use try-catch for exception handling.

Reply:
address as discussed in the huddle.